xarray/backends/common.py

```diff
@@ -391,7 +458,7 @@ def ensure_open(self, autoclose):
 
         Use requires `autoclose=True` argument to `open_mfdataset`.
         """
-        if self._autoclose and not self._isopen:
+        if autoclose and not self._isopen:
```

Comment:
I'm not sure, but this may have been a bug. We might consider some additional logic to handle what to do when `self._autoclose` and `autoclose` conflict.